src/Lucene.Net/Search/FieldCacheImpl.cs

```diff
@@ -193,57 +191,31 @@ internal Cache(FieldCache wrapper)
             
             internal FieldCache wrapper;
 
-            internal WeakDictionary<object, IDictionary<Entry, object>> readerCache = new WeakDictionary<object, IDictionary<Entry, object>>();
+            internal ConditionalWeakTable<object, ConcurrentDictionary<Entry, object>> readerCache = new ConditionalWeakTable<object, ConcurrentDictionary<Entry, object>>();
```

Comment:
what are the keys here? The first is by the reader, the inner one's key is?

Reply:
The fields.